### PR TITLE
fix(pyroscope/scrape): add support for scraping `alloy`'s internal domain

### LINKED ISSUE #603: pyroscope.scrape does not support scraping `alloy.internal:12345`
### What's wrong?

`alloy.internal:12345` (default value of `--server.http.memory-addr`) is the address of the in-memory HTTP listener, used for collecting `prometheus.exporter` components without using the network. 

`alloy.internal:12345` is useful for self-scraping in environments where the real HTTP address may have been changed, or if TLS is configured in the `http` configuration block.

`pyroscope.scrape` should use the custom dial function exposed by the HTTP service [just as `prometheus.scrape` does](https://github.com/grafana/alloy/blob/3201389252d2c011bee15ace0c9f4cdbcb978f9f/internal/component/prometheus/scrape/scrape.go#L163-L165).

### Steps to reproduce

1. Run `pyroscope.scrape` with a target to scrape `alloy.internal:12345`. 
2. Observe scrape failures in logs. 

### System information

N/A

### Software version

v1.0

### Configuration

```text
pyroscope.scrape "default" {
  targets = [
    {"__address__" = "alloy.internal:12345", "service_name" = "alloy"},
  ]
  forward_to = []
}
```

### Logs

```text
.internal:12345\", service_name=\"alloy\"}" err="Get \"http://alloy.internal:12345/debug/pprof/goroutine\": dial tcp: lookup alloy.internal: no such host"
ts=2024-04-18T17:05:31.898712Z level=error msg="fetch profile failed" component_path=/ component_id=pyroscope.scrape.default scrape_pool=pyroscope.scrape.default target="{instance=\"alloy.internal:12345\", service_name=\"alloy\"}" err="Get \"http://alloy.internal:12345/debug/pprof/allocs\": dial tcp: lookup alloy.internal: no such host"
ts=2024-04-18T17:05:32.591972Z level=error msg="fetch profile failed" component_path=/ component_id=pyroscope.scrape.default scrape_pool=pyroscope.scrape.default target="{instance=\"alloy.internal:12345\", service_name=\"alloy\"}" err="Get \"http://alloy.internal:12345/debug/pprof/block\": dial tcp: lookup alloy.internal: no such host"
ts=2024-04-18T17:05:33.176142Z level=error msg="fetch profile failed" component_path=/ component_id=pyroscope.scrape.default scrape_pool=pyroscope.scrape.default target="{instance=\"alloy.internal:12345\", service_name=\"alloy\"}" err="Get \"http://alloy.internal:12345/debug/pprof/mutex\": dial tcp: lookup alloy.internal: no such host"
ts=2024-04-18T17:05:35.898323Z level=error msg="fetch profile failed" component_path=/ component_id=pyroscope.scrape.default scrape_pool=pyroscope.scrape.default target="{instance=\"alloy.internal:12345\", service_name=\"alloy\"}" err="Get \"http://alloy.internal:12345/debug/pprof/profile?seconds=14\": dial tcp: lookup alloy.internal: no such host"
```

### PR DESCRIPTION
**Pull request by hainenber**
#### PR Description

The error logs are no longer there and Alloy continues to scrape profile from `alloy.internal:12345`. Ran with config
```
pyroscope.scrape "default" {
  targets = [
    {"__address__" = "alloy.internal:12345", "service_name" = "alloy"},
  ]
  forward_to = []
}
```

image

#### Which issue(s) this PR fixes

Fixes #603 

#### PR Checklist

- [x] CHANGELOG.md updated

### CODE REVIEW

**Review by tpaschalis — APPROVED** [review]
Makes sense to me, thanks! cc @simonswine @korniltsev for awareness.

File: internal/component/pyroscope/scrape/manager.go [hunk 1]
```diff
@@ -27,17 +30,18 @@ type Manager struct {
 	triggerReload chan struct{}
 }
 
-func NewManager(appendable pyroscope.Appendable, logger log.Logger) *Manager {
+func NewManager(o []config_util.HTTPClientOption, appendable pyroscope.Appendable, logger log.Logger) *Manager {
```
Comment (simonswine) [inline internal/component/pyroscope/scrape/manager.go:33]: I would prefer we define a struct `Options` a bit like [`prometheus/scrape`](https://github.com/prometheus/prometheus/blob/d15869af32f918b9a958c3a73144f4834c935a75/scrape/manager.go#L85-L86) does.

For now it is good enough for that struct to only have that single field of `HTTPClientOptions`

**Review by simonswine — COMMENTED** [review]
Thank you for picking this task up @hainenber.

I tested it locally and it works, I have one suggestion inline and also the tests are currently broken, as the http.Service is not those.

Once that's done, it is ready to go 🙂

File: internal/component/pyroscope/scrape/manager.go (hunk 1 above)
Comment (hainenber) [reply internal/component/pyroscope/scrape/manager.go]: Done in HEAD commit :D

**Review by simonswine — APPROVED** [review]
LGTM

**Pull request merged**